crates/bevy_ui/src/widget/text.rs

```diff
@@ -1,49 +1,204 @@
+    MeasureArgs, Node, NodeMeasure, Style, TargetCamera, UiScale, ZIndex,
 };
 use bevy_asset::Assets;
+use bevy_derive::{Deref, DerefMut};
 use bevy_ecs::{
+    change_detection::DetectChanges,
     entity::{Entity, EntityHashMap},
-    prelude::{Component, DetectChanges},
+    prelude::Component,
     query::With,
     reflect::ReflectComponent,
     system::{Local, Query, Res, ResMut},
     world::{Mut, Ref},
 };
 use bevy_math::Vec2;
 use bevy_reflect::{std_traits::ReflectDefault, Reflect};
-use bevy_render::{camera::Camera, texture::Image};
+use bevy_render::{camera::Camera, texture::Image, view::Visibility};
 use bevy_sprite::TextureAtlasLayout;
 use bevy_text::{
-    scale_value, CosmicBuffer, CosmicFontSystem, Font, FontAtlasSets, JustifyText, LineBreak,
-    SwashCache, Text, TextBounds, TextError, TextLayoutInfo, TextMeasureInfo, TextPipeline,
-    YAxisOrientation,
+    scale_value, ComputedTextBlock, CosmicFontSystem, Font, FontAtlasSets, LineBreak, SwashCache,
+    TextBlock, TextBounds, TextError, TextLayoutInfo, TextMeasureInfo, TextPipeline, TextReader,
+    TextSpanAccess, TextStyle, TextWriter, YAxisOrientation,
 };
+use bevy_transform::components::Transform;
 use bevy_utils::{tracing::error, Entry};
 use taffy::style::AvailableSpace;
 
-/// Text system flags
+/// UI text system flags.
 ///
 /// Used internally by [`measure_text_system`] and [`text_system`] to schedule text for processing.
 #[derive(Component, Debug, Clone, Reflect)]
 #[reflect(Component, Default, Debug)]
-pub struct TextFlags {
-    /// If set a new measure function for the text node will be created
-    needs_new_measure_func: bool,
-    /// If set the text will be recomputed
+pub struct TextNodeFlags {
+    /// If set then a new measure function for the text node will be created.
+    needs_measure_fn: bool,
+    /// If set then the text will be recomputed.
     needs_recompute: bool,
 }
 
-impl Default for TextFlags {
+impl Default for TextNodeFlags {
     fn default() -> Self {
         Self {
-            needs_new_measure_func: true,
+            needs_measure_fn: true,
             needs_recompute: true,
         }
     }
 }
 
+/// The top-level UI text component.
+///
+/// Adding `TextNEW` to an entity will pull in required components for setting up a UI text node.
+///
+/// The string in this component is the first 'text span' in a hierarchy of text spans that are collected into
+/// a [`TextBlock`]. See [`TextSpan`] for the component used by children of entities with `TextNEW`.
+///
+/// Note that [`Transform`] on this entity is managed automatically by the UI layout system.
+///
+/*
+```
+# use bevy_asset::Handle;
+# use bevy_color::Color;
+# use bevy_color::palettes::basic::BLUE;
+# use bevy_ecs::World;
+# use bevy_text::{Font, JustifyText, TextBlock, TextStyle};
+# use bevy_ui::Text;
+#
+# let font_handle: Handle<Font> = Default::default();
+# let mut world = World::default();
+#
+// Basic usage.
+world.spawn(Text::new("hello world!"));
```

Comment:
I'm assuming that `TextNEW` is some sort of temporary name, since here you show it as `Text`?